kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/BankParametersAccessServiceImpl.java

```diff
@@ -0,0 +1,759 @@
+            vaCustRefNumSet = getValueSetForParameter(KFSConstants.BankTransactionsParameters.VA_CUST_REF_NUM_SET, getVaDelimiter());
+        }
+        return vaCustRefNumSet;
+    }
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getVaNotesSixRegex()
+     */
+    public String getVaNotesSixRegex() {
+        if (vaNotesSixRegex == null) {
+            vaNotesSixRegex = parameterService.getParameterValueAsString(DocumentCreationStep.class, KFSConstants.BankTransactionsParameters.VA_NOTES_SIX_REGEX);
+        }
+        return vaNotesSixRegex;
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getVaNotesSixPattern()
+     */
+    public Pattern getVaNotesSixPattern() {
+        if (vaNotesSixPattern == null) {
+            vaNotesSixPattern = Pattern.compile(getVaNotesSixRegex());
+        }
+        return vaNotesSixPattern;
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getVaNotesSevenPattern()
+     */
+    public Pattern getVaNotesSevenPattern() {
+        if (vaNotesSevenPattern == null) {
+            vaNotesSevenPattern = Pattern.compile(getVaNotesSevenRegex());
+        }
+        return vaNotesSevenPattern;
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getVaNotesSevenMaskingPattern()
+     */
+    public Pattern getVaNotesSevenMaskingPattern() {
+        if (vaNotesSevenMaskingPattern == null) {
+            vaNotesSevenMaskingPattern = Pattern.compile(getVaNotesSevenMaskingRegex());
+        }
+        return vaNotesSevenMaskingPattern;
+    }
+
+
+    public String getVaNotesSevenRegex() {
+        if (vaNotesSevenRegex == null) {
+            vaNotesSevenRegex = parameterService.getParameterValueAsString(DocumentCreationStep.class, KFSConstants.BankTransactionsParameters.VA_NOTES_SEVEN_REGEX);
+        }
+        return vaNotesSevenRegex;
+    }
+
+    public String getVaNotesSevenMaskingRegex() {
+        if (vaNotesSevenMaskingRegex == null) {
+            vaNotesSevenMaskingRegex = parameterService.getParameterValueAsString(DocumentCreationStep.class, KFSConstants.BankTransactionsParameters.VA_NOTES_SEVEN_MASKING_REGEX);
+        }
+        return vaNotesSevenMaskingRegex;
+    }
+
+
+    /*
+   * Several of the parameters are delimited strings that should be
+   * placed into a data structure. A set was chosen, since all of
+   * the pulled values within one parm should be unique. Also, this
+   * allows us to use Set.contains(...) rather than looping through
+   * ourselves.
+   */
+    private Set<String> getValueSetForParameter(String parameterKey, String delimiter) {
+        String parameter = parameterService.getParameterValueAsString(DocumentCreationStep.class, parameterKey);
+        String[] listAsArray = parameter.split(delimiter);
+        Set<String> resultSet = new HashSet<String>();
+        for (String result : listAsArray) {
```

Comment:
No change requested: this for loop can be replaced with a one-liner of Collections.addAll(listAsArray).